services/repository-managers/src/main/java/org/sagebionetworks/repo/manager/util/UserAccessRestrictionUtils.java

```diff
@@ -0,0 +1,35 @@
+package org.sagebionetworks.repo.manager.util;
+
+import org.sagebionetworks.repo.model.ar.UsersRequirementStatus;
+import org.sagebionetworks.repo.model.ar.UsersRestrictionStatus;
+import org.sagebionetworks.repo.model.dbo.entity.UserEntityPermissionsState;
+import org.sagebionetworks.util.ValidateArgument;
+
+public class UserAccessRestrictionUtils {
+
+    public static boolean doesUserHaveUnmetAccessRestrictionsForEntity(UserEntityPermissionsState userEntityPermissionsState,
+                                                                       UsersRestrictionStatus usersRestrictionStatus) {
+        ValidateArgument.required(userEntityPermissionsState, "userEntityPermissionsState");
```

Comment:
some cases just need to know if there is any unmet.  Other cases need to know what AR IDs are met.  Both cases can be served by returning a list of unmet AR ids (Collections.emtpty() when no unmet).